packages/viz/src/lib/style/Style.ts

```diff
@@ -1,17 +1,32 @@
 import { GeoJsonLayerProps } from '@deck.gl/layers/geojson-layer/geojson-layer';
+import { Source } from '../sources/Source';
+
+export type StyleProperties =
+  | GeoJsonLayerProps<any>
+  | ((source: Source) => GeoJsonLayerProps<any>);
 
 export class Style {
-  private _styleProperties: GeoJsonLayerProps<any>;
+  private _styleProperties: StyleProperties;
+  private _field?: string;
 
-  constructor(styleProperties: GeoJsonLayerProps<any>) {
-    this._styleProperties = { ...styleProperties };
+  constructor(styleProperties: StyleProperties, field?: string) {
+    this._styleProperties = styleProperties;
+    this._field = field;
   }
 
-  public hasProperties() {
-    return Boolean(Object.keys(this._styleProperties).length);
-  }
+  public getProperties(source?: Source) {
```

Comment:
I think that we could extend the mechanism, maybe using the whole `(layer?: Layer)` as parameter, to allow a more open access to other pieces of info for the style functions. 

Then we could have, internally calls like
`layer.getSource` for stats-related or `layer.getMap().viewport` for zoom-related styles

Reply:
I've tried to do it but if we do that we create a circular dependency between Layer and Styles

Reply:
Maybe we can define a new *ILayer* interface, implemented by *Layer*, and make both pieces dependent on it to break the cycle